python/lib/sift_py/upload/status.py

```diff
@@ -0,0 +1,61 @@
+import re
+import time
+from urllib.parse import urljoin
+
+import requests
+from sift_py.rest import SiftRestConfig
+
+
+class DataImportStatus:
+    STATUS_PATH = "/api/v1/data-imports"
+    _data_import_id: str
+
+    def __init__(self, restconf: SiftRestConfig, data_import_id: str):
+        base_uri = self.__class__._compute_uri(restconf)
+        self._data_import_id = data_import_id
+        self._status_uri = urljoin(base_uri, self.STATUS_PATH)
+        self._apikey = restconf["apikey"]
+
+    def get_status(self):
+        response = requests.get(
+            url=f"{self._status_uri}/{self._data_import_id}",
+            headers={"Authorization": f"Bearer {self._apikey}"},
+        )
+        response.raise_for_status()
+        return response.json().get("dataImport").get("status")
+
+    def wait(self, verbose: bool = False) -> bool:
```

Comment:
I may have been the one to suggest printing but I think we should change this a tad. Rather than printing I think the current implementation makes it quite easy for users to do this on their own. Rather than call it `wait`, I think something more descriptive like `wait_until_complete` or something like that would be good with no additional args other than `self`. Then inside the function we'll poll, and if the status is complete it would be nice if `wait` returned the upload_url and/or upload ID of the file that was uploaded.